stellargraph/layer/knowledge_graph.py

```diff
@@ -175,27 +175,93 @@ def rank_edges_against_all_nodes(model, test_data, known_edges_graph):
 
         num_nodes = known_edges_graph.number_of_nodes()
 
-        def ranks(pred, true_ilocs):
-            batch_size = len(true_ilocs)
+        def ranks(pred, true_node_ilocs, true_rel_ilocs, modified_object):
+            batch_size = len(true_node_ilocs)
             assert pred.shape == (num_nodes, batch_size)
 
             # the score of the true edge, for each edge in the batch (this indexes in lock-step,
-            # i.e. [pred[true_ilocs[0], range(batch_size)[0]], ...])
-            true_scores = pred[true_ilocs, range(batch_size)]
+            # i.e. [pred[true_node_ilocs[0], range(batch_size)[0]], ...])
+            true_scores = pred[true_node_ilocs, range(batch_size)]
 
             # for each column, compare all the scores against the score of the true edge
             greater = pred > true_scores
 
             # the raw rank is the number of elements scored higher than the true edge
             raw_rank = 1 + greater.sum(axis=0)
-            assert raw_rank.shape == (batch_size,)
 
-            return raw_rank
+            # FIXME(#870): this would be better without external IDs <-> ilocs translation
+            true_nodes = known_edges_graph._nodes.ids.from_iloc(true_node_ilocs)
+            true_rels = known_edges_graph._edges.types.from_iloc(true_rel_ilocs)
+            for i, (node_id, rel) in enumerate(zip(true_nodes, true_rels)):
+                if modified_object:
+                    neighs = known_edges_graph.out_nodes(node_id, edge_types=[rel])
```

Comment:
previously, if `modified_object` (or `true_is_source`) was `True`, I think it was the in edges being calculated:

```
real_edges = [
    (s, i)
    for i, (obj, rel) in enumerate(zip(true_ilocs, rels))
    for s, o in edges_by_type[rel] if o == obj
]
```
because we have `o`, and we're finding `s->o`. but... I think its a bug here that we're passing the `object` indices, 

When `modified_object` is True:

- We *should* be passing the `subject` indices and finding the out edges
- My code was passing the `object` indices and finding the in edges
- Your code is passing the `object` indices and finding the out edges

If that makes sense?